extract.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+# Authors:
+# 2017 Youfu Zhang
+# 2019 Balint Reczey <balint.reczey@canonical.com>
+
+set -e
+
+usage() {
+    echo "Usage: $0 < Unifi Controller backup file > < output directory >"
+}
+
+if [ -z "$2" -o ! -f "$1" ]; then
+    usage
+    exit 1
+fi
+
+INPUT="$1"
+OUTPUT="$2"
+
+mkdir -p "${OUTPUT}"
+
+TMP_ZIP1=$(mktemp --suffix=.zip)
+TMP_ZIP2=$(mktemp --suffix=.zip)
+
+trap "rm -f ${TMP_ZIP1} ${TMP_ZIP2}" EXIT
+
+openssl enc -d -in "${INPUT}" -out "${TMP_ZIP1}" -aes-128-cbc -K 626379616e676b6d6c756f686d617273 -iv 75626e74656e74657270726973656170 -nosalt -nopad
```

Comment:
I think `-nosalt` may not be needed, since key and salt is already provided. Not tested though.